src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagJobs.java

```diff
@@ -17,36 +28,137 @@
         Scheduler.scheduleJob(
             new BugsnagEventHandlingJob(),
-            BUGSNAG_EVENT_REQUEST_JOB_DELAY_IN_MINUTES + BUGSNAG_EVENT_JOB_DELAY_IN_MINUTES,
+            0,
             BUGSNAG_EVENT_JOB_DELAY_IN_MINUTES,
             TimeUnit.MINUTES);
     }
+
+    /**
+     * Schedule event request jobs.
+     */
+    private static void scheduleEventRequestJob(Set<String> projectIds, int initialDelay) {
+        Scheduler.scheduleJob(
+            new BugsnagEventRequestJob(projectIds),
+            initialDelay,
+            BUGSNAG_EVENT_REQUEST_JOB_DELAY_IN_HOURS,
+            TimeUnit.HOURS);
+    }
+
+    /**
+     * Seed a project if the last event data request is greater than a day.
+     */
+    private static Set<String> seedDataIfNeeded() {
+        Set<String> projectsRequested = new HashSet<>();
+        Set<String> projectIds = MonitoredComponent.getComponentsByProjectId().keySet();
+        projectIds.forEach(projectId ->
+            {
+                int hoursSinceLastRequest = getHoursSinceLastRequest(projectId);
+                // Only seed data if days since is greater than one day (otherwise we're duplicating the scheduled)
+                if (hoursSinceLastRequest > BUGSNAG_EVENT_REQUEST_JOB_DELAY_IN_HOURS) {
+                    LOG.debug("Triggered Bugsnag event request for project {}", projectId);
+                    BugsnagEventRequestJob.triggerEventDataRequestForProject(projectId, hoursSinceLastRequest / 24);
+                    projectsRequested.add(projectId);
+                }
+            }
+        );
+        return projectsRequested;
+    }
+
+    /**
+     * Define how long in hours since a project's last event data request took place. Restrict to
+     * {@link BugsnagDispatcher#BUGSNAG_REPORTING_WINDOW_IN_DAYS} (in hours) if greater than this.
+     */
+    public static int getHoursSinceLastRequest(String projectId) {
+        int hoursSinceLastRequest = 0;
+        // Get latest successful request for project.
+        BugsnagEventRequest compareRequest = getLatestComparisonRequest(projectId);
+        if (compareRequest != null) {
+            // Otherwise, determine how many hours have passed since the end of the last request's time window.
+            hoursSinceLastRequest = getHoursRoundedToWholeDaySinceDate(compareRequest.getTimeWindowEndInMillis());
+        }
+        // Restrict the hours since last request if greater than the reporting window.
+        return (compareRequest != null)
+            ? Math.min(hoursSinceLastRequest, BUGSNAG_REPORTING_WINDOW_IN_DAYS * 24)
+            : BUGSNAG_REPORTING_WINDOW_IN_DAYS * 24;
+    }
+
+    /**
+     * Return the number of hours (rounded to a whole day) that have passed since the date provided.
+     */
+    public static int getHoursRoundedToWholeDaySinceDate(long historicDateInMillis) {
+        long reportingGapMillis = new Date().getTime() - historicDateInMillis;
```

Comment:
Please use DateTimeUtils in case we want to mock the time in a test in the future.

```suggestion
        long reportingGapMillis = DateTimeUtils.currentTimeMillis() - historicDateInMillis;
```